infor-design/enterprise-wc
# 1205 - IdsPopupMenu:  Enable scrolling in fixed-sized `max-height` menus

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR allows IdsPopupMenu configured with a `max-height` property to scroll its content.  This scrolling behavior also accounts for submenu position, placing opened submenus adjacent to their parent menu items as expected.

Additionally, a change in the IdsToolbarMoreActions component structure was needed to account for the menu changes.  Now, IdsToolbarMoreActions allows users to slot IdsPopupMenu and IdsMenuButton elements as children (not just IdsMenuGroup elements).  A codepath for "migration" from the old slotting method has been added, and tests and documentation have been updated to reflect this change.

**Related github/jira issue (required)**:
Closes #1205

**Steps necessary to review your pull request (required)**:
Pull/build/run, then try these test steps:

_Test scrollable menus:_
- Open http://localhost:4300/ids-popup-menu/scrolling.html
- Right click to activate a scrolling IdsPopupMenu
- Scroll down and activate submenu items.  The menu should appear adjacent to its parent menu item
- Also try this on IdsDataGridCells: http://localhost:4300/ids-data-grid/contextmenu-thru-data.html

_Test non-scrollable menus:_
Open and test the context menus and/or menu buttons in these examples.  When testing IdsPopupMenu components, and make sure their submenus open accurately:
- http://localhost:4300/ids-popup-menu/example.html
- http://localhost:4300/ids-popup-menu/selected-state.html
- http://localhost:4300/ids-popup-menu/data-driven.html
- http://localhost:4300/ids-menu-button/example.html

_Test IdsToolbarMoreActions:_
- Open http://localhost:4300/ids-toolbar/example.html
- Use the "..." button and activate its submenus.  Submenus should open in the correct place
- Shrink the browser viewport to push the "Menu" button into toolbar overflow.  Use the "..." and activate the "Menu" item's submenu, and its child submenus.  All elements should display within the viewport.

**Included in this Pull Request**:
